execute/plugin.go

```diff
@@ -620,23 +626,36 @@ func (p *Plugin) checkAlreadyExecuted(
 ) error {
 	seqNrRangesBySource := getSeqNrRangesBySource(reports)
 
-	// TODO: batch these queries? these are all DB reads.
-	// maybe some alternative queries exist.
-	for sourceChainSelector, seqNrRange := range seqNrRangesBySource {
-		executed, err := p.ccipReader.ExecutedMessages(ctx, sourceChainSelector, seqNrRange, primitives.Unconfirmed)
-		if err != nil {
-			return fmt.Errorf("couldn't check if messages already executed: %w", err)
+	executed, err := p.ccipReader.ExecutedMessages(
+		ctx,
+		seqNrRangesBySource,
+		primitives.Unconfirmed)
+	if err != nil {
+		return fmt.Errorf("couldn't check if messages already executed: %w", err)
+	}
+
+	for sourceChainSelector, expectedSqNrRanges := range seqNrRangesBySource {
+		snRangeSet := mapset.NewSet[cciptypes.SeqNum]()
+		totalSeqNums := 0
+		for _, snRange := range expectedSqNrRanges {
+			snRangeSet.Append(snRange.ToSlice()...)
+			totalSeqNums += snRange.Length()
 		}
```

Comment:
Can we move this below the `if _, exists` if statement which breaks the loop early?